code/modules/nifsoft/nif_tgui.dm

```diff
@@ -15,15 +15,67 @@
 		"retro",
 		"syndicate"
 	)
-	var/tmp/obj/effect/statclick/nif_open/our_statclick
 	var/tmp/last_notification
+	var/tmp/datum/component/nif_menu/menu
 
 /**
- * Special stat button for the interface
+ * Small helper component to manage the HUD icon
  */
-/obj/effect/statclick/nif_open
-/obj/effect/statclick/nif_open/Click(location, control, params)
-	var/obj/item/device/nif/N = target
+/datum/component/nif_menu
+	var/obj/screen/nif/screen_icon
+
+/datum/component/nif_menu/Initialize()
+	. = ..()
+	if(ismob(parent))
+		RegisterSignal(parent, COMSIG_MOB_CLIENT_LOGIN, .proc/create_mob_button)
+		var/mob/owner = parent
+		if(owner.client)
+			create_mob_button(parent)
+
+/datum/component/nif_menu/Destroy()
+	if(ishuman(parent))
+		var/mob/living/carbon/human/H = parent
+		H.verbs -= /mob/living/carbon/human/proc/nif_menu
+		if(screen_icon)
+			H.client?.screen -= screen_icon
+	qdel_null(screen_icon)
+	return ..()
```

Comment:
```suggestion
/datum/component/nif_menu/Initialize()
	if(!ismob(parent))
		return COMPONENT_INCOMPATIBLE
	. = ..()

/datum/component/nif_menu/RegisterWithParent()
	. = ..()
	RegisterSignal(parent, COMSIG_MOB_CLIENT_LOGIN, .proc/create_mob_button)
	var/mob/owner = parent
	if(owner.client)
		create_mob_button(parent)

/datum/component/nif_menu/UnregisterFromParent()
	. = ..()
	UnregisterSignal(parent, COMSIG_MOB_CLIENT_LOGIN)
	if(ismob(parent))
		var/mob/owner = parent
		if(screen_icon)
			owner?.client?.screen -= screen_icon
			UnregisterSignal(screen_icon, COMSIG_CLICK)
			qdel_null(screen_icon)
		if(ishuman(parent))
			owner.verbs -= /mob/living/carbon/human/proc/nif_menu
		
```

Reply:
Not *super* important because this isn't a transferable component, but it's better to use the register/unregister procs for interacting with the parent